src/material-examples/cdk-datepicker-restricting-available-dates/cdk-datepicker-restricting-available-dates-example.ts

```diff
@@ -0,0 +1,137 @@
+import {Component, Input} from '@angular/core';
+import {CalendarView, DateAdapter} from '@angular/cdk/datetime';
+
+/** @title CDK Datepicker with filter, min and max, and start date validation */
+@Component({
+  selector: 'cdk-datepicker-restricting-available-dates-example',
+  templateUrl: 'cdk-datepicker-restricting-available-dates-example.html',
+  styleUrls: ['cdk-datepicker-restricting-available-dates-example.css'],
+})
+export class CdkDatepickerRestrictingAvailableDatesExample {
+  dates: Date[] = [];
+  minDate = new Date(2000, 0, 1);
+  maxDate = new Date(2020, 0, 1);
+  startDate = new Date(1990, 0, 1);
+
+  constructor() {
+    this.dates.push(new Date(1800,8,9));
+    this.dates.push(new Date(1999,8,8));
+    this.dates.push(new Date(2018,8,15));
+  }
+  myFilter = (d: Date): boolean => {
+    if (d) {
+      const day = d.getDay();
+      // Prevent Saturday and Sunday from being selected.
+      return day !== 0 && day !== 6;
+    } else {
+      return true;
+    }
+  }
+}
+
+
+@Component({
+  selector: 'my-filter-calendar',
+  outputs: ['selectedChange'],
+  template: `
+    <div *ngFor="let date of dates">
+      <button (click)="_selected(date)">{{date}}</button>
+    </div>
+    <div>{{this.validDate}}</div>
```

Comment:
these examples seem kind of weird, why do we show a message saying the date is invalid instead of just disabling the button?

Reply:
I now disable the buttons